### PR TITLE
Add Knights

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://invent.kde.org/games/knights/-/issues/1
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* https://invent.kde.org/games/knights/-/merge_requests/7

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build org.kde.knights

**Comment by hfiguiere** [comment]
cc @flathub/kde

File: org.kde.knights.json [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        "/share/man",
+        "/include"
+    ],
+    "modules": [
+        {
+            "name": "gnuchess",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://ftpmirror.gnu.org/chess/gnuchess-6.2.5.tar.gz",
```
Comment (tsdgeos) [inline org.kde.knights.json:26]: Use 6.2.9 ?

File: org.kde.knights.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            "name": "knights",
+            "buildsystem": "cmake-ninja",
+            "post-install": [
+                "install -Dm644 org.kde.knights.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/org.kde.knights.svg"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://invent.kde.org/games/knights.git",
+                    "tag": "v22.04.1",
```
Comment (tsdgeos) [inline org.kde.knights.json:52]: use tarball

**Review by tsdgeos — CHANGES_REQUESTED** [review]

**Comment by travier** [comment]
Some flatpak external data checker config would be great but otherwise manifest LGTM.

**Comment by Mailaender** [comment]
https://release-monitoring.org/project/18814/ is watching GitHub which I was not allowed to use as a source.

**Comment by tsdgeos** [comment]
You can use the external data checker used for tarballs that dozens of KDE applications use.

**Comment by cho2** [comment]
> https://release-monitoring.org/project/18814/ is watching GitHub which I was not allowed to use as a source.

Use https://release-monitoring.org/project/8763/

**Comment by Mailaender** [comment]
Added

**Comment by tsdgeos** [comment]
bot, build org.kde.knights

File: org.kde.knights.json [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.15-21.08",
+    "sdk": "org.kde.Sdk",
+    "command": "knights",
+    "rename-icon": "knights",
+    "finish-args": [
+        "--device=dri",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland"
```
Comment (tsdgeos) [inline org.kde.knights.json:12]: don't we need network so you can play in freechess.org and pulseaudio so that sound works?

Comment (Mailaender) [reply org.kde.knights.json]: Added

**Comment by Mailaender** [comment]
bot, build org.kde.knights

**Pull request closed without merging**